tools/websocktunnel/wsproxy/stopper.go

```diff
@@ -0,0 +1,46 @@
+package wsproxy
+
+import "sync"
+
+// stopper manages stopping a fleet of goroutines.  It is basically
+// a protected boolean with functionality to set, check, and wait.
```

Comment:
Is this a common thing we do in our go code? Worth moving something like this to `internal`? (not blocking landing this on that)

Reply:
I don't know.. this is the first time I've built something specifically like this.  If we see it again, yeah, it would probably make a good utility.